File: Lab1/Python/func.py
# Введення файлу через консоль
def inputfile(path, num):
    file = open(path, "w")
    string = input("Введіть текст " + str(num) + "-го файлу:\n")
    while chr(1) not in string: #Ознака кінця введення файлу Ctrl+A
        file.write(string+'\n')
        string = input()
    file.close()

# Створення вихідного файлу
def croutfile(path1, path2, path3):  
    outf = open(path3, 'w')  
    for i in matchlines(path1, path2):
        outf.write(i)  
    outf.close()

# Пошук однакових рядків
def matchlines(path1, path2): 
    inpf1 = lstlines(path1)
    inpf2 = lstlines(path2)
    lst=[]
    for i in inpf1:
        if i not in inpf2:
            lst.append(i)
    return lst

# Створення списка з файлу
def lstlines(path):
    file = open(path)
    lst = file.readlines()
    file.close()
    return lst

# Визначення кількості рядків у файлі
def filelen(path):
    with open(path) as file:
        return len(file.readlines())

# Вивід файлу в консоль
def printfile(path): 
    print("Зміст файлу", path.split("/")[-1:][0])
    lst = lstlines(path)
    if len(lst)!=0:
        for i in lst:
            print(i, end="")
    else:
        print("Файл пустий")

File: Lab1/C++/func.cpp
#include "func.h"

void inputfile(string path, int num)
{
    string str;
    ofstream file(path);
    cout << "Введіть текст " << num << "-го файлу:" << endl;
    getline(cin, str);
    while(str[str.length()-1] != 1) //Ознака кінця введення файлу Ctrl+A
    {
        file << str << endl;
        getline(cin, str);
    }
    file.close();
}

void croutfile(string path1, string path2, string path3)
{
    ofstream outf(path3);
    vector<string> lst = matchlines (path1, path2);
    for (int i = 0; i < lst.size(); i++) outf << lst[i] << endl;
    outf.close();
}  

vector<string> matchlines(string path1, string path2)
{
    vector<string> inpf1 = lstlines(path1);
    vector<string> inpf2 = lstlines(path2);
    vector<string> lst;
    for (int i = 0; i < inpf1.size(); i++)
    {
        if (find(inpf2.begin(), inpf2.end(), inpf1[i])==inpf2.end()) lst.push_back(inpf1[i]);
    }
    return lst;
}

vector<string> lstlines(string path)
{
    ifstream file(path);
    string str;
    vector<string> lst;
    while(getline(file, str)) lst.push_back(str);
    file.close();
    return lst;
}

int filelen(string path)
{
    int res = 0;
    string temp;
    ifstream file(path);
    while(getline(file, temp)) res++;
    file.close();
    return res;
}

void printfile(string path)
{
    cout << "Зміст файлу " << path << endl;
    vector<string> lst = lstlines(path);
    if (lst.size()!=0)
    {
        for (int i = 0; i < lst.size(); i++) cout << lst[i] << endl;
    }
    else cout << "Файл пустий" << endl;
}